File: projects/fizzbuzz/kotlin/trent/Fizzbuzz.kt
fun fizzbuzz(max: Int) {
    for (i in 1 until max + 1) {
        println(checkFizz(i))
    }
}

fun checkFizz(input: Int): String {
    var output: String = ""

    if (input % 3 == 0) output += "Fizz"
    if (input % 5 == 0) output += "Buzz"
    if (input % 7 == 0) output += "Bazz"
    if (output.length == 0) output += input.toString()

    return output
}

fun testCheckFizz() {
    assert(checkFizz(1) == "1")
    assert(checkFizz(3) == "Fizz")
    assert(checkFizz(5) == "Buzz")
    assert(checkFizz(7) == "Bazz")
    assert(checkFizz(15) == "FizzBuzz")
    assert(checkFizz(21) == "FizzBazz")
    assert(checkFizz(35) == "BuzzBazz")
    assert(checkFizz(105) == "FizzBuzzBazz")
}

fun main(args: Array<String>) {
    testCheckFizz()
    try {
        var input = args[0].toInt()
        if (input < 1) throw Exception("ERROR: Invalid input")
        println("Starting Fizzbuzz on $input")
        fizzbuzz(input)
    } catch (e: Exception) {
        println("Please provide a positive non-zero integer as an argument")
    }
}

File: projects/fizzbuzz/python/Lightner/FizzBuzz.py

def fizzbuzz(num):
    if num % 3 == 0 and num % 5 == 0 and num % 7 == 0:
        return('FizzBuzzBazz')
    elif num % 3 == 0 and num % 5 == 0:
        return('FizzBuzz')
    elif num % 3 == 0 and num % 7 == 0:
        return('FizzBazz')
    elif num % 5 == 0 and num % 7 == 0:
        return('BuzzBazz')
    elif num % 3 == 0:
        return('Fizz')
    elif num % 5 == 0:
        return('Buzz')
    elif num % 7 == 0:
        return('Bazz')
    else:
        return(num)
        


while True:
    try:
        n = int(input("Enter number: "))
        break
    except ValueError:
        print("Invalid input. Try again.")
    
        
        
if n < 1:
    print('Not a positive integer')
for number in range(1, n):
    print (fizzbuzz(number))
